src/frontend/web_application/src/services/message/index.js

```diff
@@ -9,3 +9,13 @@ export const sortMessages = (messages, reversed) => messages.sort((a, b) => {
 export const renderParticipant = participant => `${participant.label} (${participant.address})`;
 
 export const getLastMessage = messages => sortMessages(messages, true)[0];
+
+export const getAuthor = message => message.participants && message.participants.find(participant => participant.type === 'From');
+
+export const isMessageFromUser = (message, user) => {
+  const author = getAuthor(message);
+  const authorId = author && author.contact_ids[0];
+  const userId = user && user.contact.contact_id;
+
+  return userId && authorId && userId === authorId;
```

Comment:
I think userId && authorId MUST be set, it sounds not useful to do a condition with, isn't it?

In case those are not set this probably is a bug.

Reply:
btw, a unit test would be great ;)